### PR TITLE
New Actor interface

### PR DESCRIPTION
**Pull request by sleipnir**
Guys how the PR for improvements has been long in the past so I'm bringing parts of it in "smaller" and easier to read PRs. I'm going to ask you to review it as soon as possible because I want to get on with it soon.

Note: This will break the SDKs. Being that the only one that I kept compatibility for now was the Elixir itself.

### CODE REVIEW

File: apps/spawn/priv/protos/eigr/functions/protocol/actors/actor.proto [hunk 1]
```diff
@@ -43,10 +43,39 @@ message ActorState {
...
+
+message ActorId {
+    // The name of a Actor Entity.
+    string name = 1;
+
+    // Name of a ActorSystem
+    string system = 2;
+
+    // Actor identifier in persistent storage. For internal proxy use only
+    string hash_key = 3;
```
Comment (marcellanz) [inline apps/spawn/priv/protos/eigr/functions/protocol/actors/actor.proto:69]: perhaps we can offset internal fields? with evolution for public field we would not mix them?

Comment (sleipnir) [reply apps/spawn/priv/protos/eigr/functions/protocol/actors/actor.proto:69]: > perhaps we can offset internal fields? with evolution for public field we would not mix them?

I agree. We can remove it here

**Comment by eliasdarruda** [comment]
This will also break internal test factories, we can work in a different PR to solve this

**Review by eliasdarruda — APPROVED** [review]

File: apps/spawn/priv/protos/eigr/functions/protocol/actors/actor.proto (hunk 1 above)
Comment (marcellanz) [reply apps/spawn/priv/protos/eigr/functions/protocol/actors/actor.proto]: I meant, offset with the proto field numbers; not removing it. I think its fine to have fields if needed internally. Alas, I'm not sure if we could even have an proxy-internal version without the "private" field, with not colliding proto field by their id.

Comment (sleipnir) [reply apps/spawn/priv/protos/eigr/functions/protocol/actors/actor.proto]: I decided to remove it while I study this further. We can include if necessary in another PR.

**Review by marcellanz — APPROVED** [review]
LGTM, thanks.

**Pull request merged**